### PR TITLE
Bump to Go 1.21

### PR DESCRIPTION
**Pull request by ArangoGutierrez**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -8,7 +8,7 @@ IMAGE_BUILD_CMD ?= docker build
 IMAGE_BUILD_EXTRA_OPTS ?=
 IMAGE_PUSH_CMD ?= docker push
 CONTAINER_RUN_CMD ?= docker run
-BUILDER_IMAGE ?= golang:1.20-bullseye
+BUILDER_IMAGE ?= golang:1.21-bullseye
```
Comment (mythi) [inline Makefile:11]: not important but could move to Debian 12 at the same time?
```suggestion
BUILDER_IMAGE ?= golang:1.21-bookworm
```

Comment (ArangoGutierrez) [reply Makefile:11]: I am ok with that, @marquiz ?

Comment (marquiz) [reply Makefile:11]: Yeah, let's do this. Follow the tip of the world on mainline :)

**Comment by marquiz** [comment]
Looks good, but wait for https://github.com/kubernetes/test-infra/pull/30940
/hold

**Review by marquiz — APPROVED** [review]
/lgtm

**Pull request merged**